sdk/identity/azure-identity/azure/identity/aio/_internal/aad_client.py

```diff
@@ -66,7 +66,7 @@ async def obtain_token_by_refresh_token(
         request = self._get_refresh_token_request(scopes, refresh_token, **kwargs)
         return await self._run_pipeline(request, **kwargs)
 
-    async def obtain_token_by_refresh_token_on_behalf_of(
+    async def obtain_token_by_refresh_token_on_behalf(
```

Comment:
Why we rename it?

Reply:
pylint now complains about method names over 40 characters long, and the `_of` makes it 42. 😞 

Since, this is internal, and the name is still clear without the `_of`, I opted to just shorten the name, but I can always add a pylint disable comment for this line.

Reply:
The concern is we use "on_behalf_of" in other methods. It is better to keep consistent.

Reply:
Reverted the name change and added the pylint disable marker to ignore the name-too-long rule .